src/goto-programs/interpreter_class.h

```diff
@@ -279,8 +280,9 @@ class interpretert:public messaget
   {
     mp_vectort v;
     evaluate(expr, v);
+    // FIXME not sure if this can happen
     if(v.size()!=1)
-      throw "invalid boolean value";
+      throw interpreter_errort("invalid boolean value");
```

Comment:
I wonder if this shouldn't be an invariant. The input `expr` is always the path condition of the interpreted program and I don't think the user is directly responsible for what the interpreter chooses to be a path condition.